### PR TITLE
Integrate git cloning and pushing

### PR DESCRIPTION
**Pull request by DylanVanAssche**
This PR adds the setup to use git cloning and pushing of your CalDAV/CardDAV files.

You need the following environment variables:

1. `GIT_REPOSITORY`: The repository URL, with username and password, example: `https://myuser:mypass@github.com/myuser/myrepo`
2. `GIT_USERNAME`: The username to be used for commits
3. `GIT_EMAIL`: The email to be used for commits

And enable the `hook` in your config file:

```
[storage]
hook = git add -A && (git diff --cached --quiet || git commit -m "Changes by "%(user)s) && git push origin master
```

This option is fully optional, if the environment variables are not defined, the git repository is not used.
Looking forward to your review :)

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -10,6 +10,15 @@ if [ -n "$UID" ]; then
     usermod -o -u "$UID" radicale
 fi
 
+# Clone Git repository if enabled
+if [ -n "$GIT_REPOSITORY" ] && [ -n "$GIT_USERNAME" ] && [ -n "$GIT_EMAIL" ]; then
+    git clone "$GIT_REPOSITORY" /data/collections
+
+    # Git needs to know an indentity for pushing changes
```
Comment (tomsquest) [inline docker-entrypoint.sh:17]: typo in `indentity`

**Review by tomsquest — CHANGES_REQUESTED** [review]
Hi @DylanVanAssche ,

Thanks for your contribution.
All seems OK :+1: .
There is one typo.

Can you add a section to the Readme to help user configure it? 
(for adding the storage section you explained in this PR).

**Comment by DylanVanAssche** [comment]
Done :+1:

**Review by tomsquest — APPROVED** [review]

**Comment by tomsquest** [comment]
@DylanVanAssche Your PR has been merged, I will publish a new version soon.
Thanks for your contribution.

**Comment by tomsquest** [comment]
Released in  `2.1.12.2`

**Pull request merged**